### DISCUSSION TITLE
GitHub CI stuck on "Uninstalling com.example.app.dev from sdk gphone64 x86 64"

### DISCUSSION THREAD

**Discussion by ghost** [leancodepl/patrol#1651] [Q&A]
Currently, we have 6 Patrol tests spinning in GitHub CI on BuildJet "buildjet-4vcpu-ubuntu-2204" runner & ReactiveCircus Android Emulator:

```
with:
api-level: 31
target: google_apis
arch: x86_64
profile: pixel_6
ram-size: 2048M
heap-size: 512M
force-avd-creation: false
emulator-options: -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
disable-animations: false
script: >
patrol test 
--target integration_test/tests/first_test.dart
--target integration_test/tests/second_test.dart
--target integration_test/tests/third_test.dart
--target integration_test/tests/fourth_test.dart
--target integration_test/tests/fifth_test.dart
--target integration_test/tests/sixth_test.dart
--flavor=$FLAVOR --dart-define=ENVIRONMENT=$FLAVOR --dart-define=E2E=true -v
```

Maybe it isn't the reason, but If it is 5 tests - everything is Ok (passed or failed - nothing stuck), but when it is 6 CI is stuck on the "Uninstalling com.example.app.dev from sdk gphone64 x86 64" (next one should be "Uninstalling com.example.app.dev.test from sdk gphone64 x86 64"). 

Also, I'm seeing this just before getting stuck on uninstalling (again, sorry if it isn't related):
```
✗ Failed to execute tests of apk with entrypoint test_bundle.dart on emulator-5554 (Gradle test execution failed with code 1) (96.0s)
Exception: Gradle test execution failed with code 1
#0      AndroidTestBackend.execute.<anonymous closure> (package:patrol_cli/src/android/android_test_backend.dart:140:9)
<asynchronous suspension>
#1      DisposeScope.run (package:dispose_scope/src/dispose_scope.dart:46:7)
<asynchronous suspension>
#2      AndroidTestBackend.execute (package:patrol_cli/src/android/android_test_backend.dart:106:5)
<asynchronous suspension>
#3      TestCommand._execute (package:patrol_cli/src/commands/test.dart:269:7)
<asynchronous suspension>
#4      TestCommand.run (package:patrol_cli/src/commands/test.dart:169:23)
<asynchronous suspension>
#5      CommandRunner.runCommand (package:args/command_runner.dart:212:13)
<asynchronous suspension>
#6      PatrolCommandRunner.runCommand (package:patrol_cli/src/runner/patrol_command_runner.dart:320:18)
<asynchronous suspension>
#7      PatrolCommandRunner.run (package:patrol_cli/src/runner/patrol_command_runner.dart:264:18)
<asynchronous suspension>
#8      patrolCommandRunner (package:patrol_cli/src/runner/patrol_command_runner.dart:76:20)
<asynchronous suspension>
#9      main (file:///opt/hostedtoolcache/flutter/stable-3.7.12-x64/.pub-cache/hosted/pub.dev/patrol_cli-2.1.2/bin/main.dart:6:20)
<asynchronous suspension>
```
Any ideas will be appreciated, TY in advance!

**Comment by fylyppo** [comment]
Can you show me whole workflow file (with nice format please)? The error message from Gradle tells that cannot run tests, maybe that's why you cannot uninstall them then.

**Comment by fylyppo** [comment]
Is there a report generated for failed run? if so please publish it somewhere and give me the link.
In the meantime you can check what happens if you remove this statement: `if: steps.gradle-cache.outputs.cache-hit != 'true'` from `Flutter build apk --debug (for Gradle Wrapper)` step.

**Reply by ghost** [reply]
1. Report isn't generated because it has "if: failure()" - when it is stuck, CI marks it as canceled on timeout. 
2. Gradle cache from "if: steps.gradle-cache.outputs.cache-hit != 'true" currently always "false" & we build an app to be able to use Gradle Wrapper.

**Reply by fylyppo** [reply]
Try to set timeout for this step to get some report

**Comment by fylyppo** [comment]
It's very interesting that 5 tests work but 6 not. Maybe there's something with this additional one but I suppose locally everything works well.

**Reply by ghost** [reply]
Yes, locally it is Ok. I'll change "if: failure()" to "if: always()" to check if anything useful will be provided in the report and let you know.

**Reply by ghost** [reply]
@fylyppo sorry I've not managed to try it out in our GitHub CI, cuz my colleagues due to active development have changed a lot in our app - so tests will need some updates. Plus, I'm vacationing till the 11th of September.

**Reply by ghost** [reply]
@fylyppo I'm back. I guess the root cause was the "AVD cache". W/o caching of emulator, everything is Ok.

**Reply by ghost** [reply]
Nope, it isn't AVD cache, it continues to get stuck with 6 tests. And "action-junit-report" tells nothing, cuz tests getting canceled by timeout.

**Reply by ghost** [reply]
For now, I'm adding the "--no-uninstall" to unblock our tests in CI.

**Reply by fylyppo** [reply]
The sixth test is the specific one of yours or you're changing it? - just to make sure that the count of tests matters here and not the specific test case.

**Reply by ghost** [reply]
It's not a specific one, it can be any. But I'm looking forward to the next stable version of Patrol, cuz it will support "group" - hope it will fix this.

**Comment by ghost** [comment]
@fylyppo can you elaborate on how exactly group() should work with latest dev versions of Patrol (2.3.0-dev.4) & patrol_cli (2.2.0-dev.2)? Can it be group() with multiple patrolTest() inside in one file? Because currently if I run "old style" (one patrolTest() per file) or if I run "new" (group() with 2 patrolTest() inside in one file) the result is the same - 1st test passed and run finished w/o launching 2nd one. Terminal shows failure, but html report shows that one (1st) test passed and that's it. TY in advance!

**Reply by ghost** [reply]
BTW this problem with stuck is gone with Flutter 3.13.0, but I'm still on Patrol 2.2.5 & patrol_cli 2.1.5 cuz newer versions have some problems.
